agent/engine/data.go

```diff
@@ -21,8 +21,75 @@ import (
 	"github.com/aws/amazon-ecs-agent/agent/utils"
 
 	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
 )
 
+// LoadState populates the task engine state with data in db.
+func (engine *DockerTaskEngine) LoadState() error {
```

Comment:
nit: could this be broken down into dedicated load* methods? for e.g. loadContainers, loadTasks etc. If it add unnecessary complexity I am not going to push it.

Reply:
sounds reasonable, i made the change